src/models/converters/chains.rs

```diff
@@ -0,0 +1,27 @@
+use crate::models::backend::chains::ChainInfo as BackendChainInfo;
+use crate::models::service::chains::{
+    ChainInfo as ServiceChainInfo, NativeCurrency as ServiceNativeCurrency, Theme as ServiceTheme,
+};
+
+impl From<BackendChainInfo> for ServiceChainInfo {
+    fn from(chain_info: BackendChainInfo) -> Self {
+        Self {
+            transaction_service: chain_info.transaction_service,
+            chain_id: chain_info.chain_id,
+            chain_name: chain_info.chain_name,
+            rpc_url: chain_info.rpc_url,
+            block_explorer_url: chain_info.block_explorer_url,
+            native_currency: ServiceNativeCurrency {
+                name: chain_info.native_currency.name,
+                symbol: chain_info.native_currency.symbol,
+                decimals: chain_info.native_currency.decimals,
+                logo_url: chain_info.native_currency.logo_url,
+            },
+            theme: ServiceTheme {
+                text_color: chain_info.theme.text_color,
+                background_color: chain_info.theme.background_color,
+            },
+            ens_registry_address: chain_info.ens_registry_address,
+        }
+    }
+}
```

Comment:
❓ If one one of the fields changes (eg.: is removed, is added) then a new type would have to be created no? Currently is mapping the type `ChainInfo` into `ChainInfo`.

Would that be done only when there's a mismatch in the payloads?

Reply:
Right, so right now they are identical. We allow for divergence now, we can remove fields in the `ServiceChainInfo` that we don't want the clients to receive. Or we can add fields to `BackendChainInfo` without having to adjust the clients each time.

We would need to adjust this mapping function depending on what we want to expose. If we add something new to `BackendChainInfo` we don't really need to change anything if we don't want it returned to the clients.

Reply:
If we remove fields from `ServiceChainInfo` in its current form then we would just need to adjust this `From`?

Reply:
Correct, and the clients would have to be notified (when this is deployed of course) since it would represent an API breaking change.